### DISCUSSION TITLE
Question: System clipboard behavior on Windows

### DISCUSSION THREAD

**Discussion by widavies** [JetBrains/ideavim#771] [Q&A] [answered]
I'm using PyCharm with IdeaVim.

My `.ideavimrc` has `clipboard+=ideaput`.

Quite often, I copy a snippet of text from the web. This links to the `*` register. Pressing `p` will paste it PyCharm. All is good.

Sometimes though, I will delete a line `dd` and try to paste with `p` only to find that the text I just deleted is pasting instead of the text from the web. The `dd` command overwrites the "*" register. So the text I copied from my browser is completely gone, I have to go back and copy it again and make sure to paste without deleting anything.

How do I fix this behavior? I want to be able to delete a few things before pasting.

(I also don't want to use the blackhole register)

**Comment by lippfi** [comment, marked as answer]
Hi! The `dd` command should not affect your system clipboard unless you have `unnamed` or `unnamedplus` in your clipboard option value.
Could you please provide more details regarding the issue:
1. What is the result of the set clipboard? command? Do you notice `unnamed` or `unnamedplus` there? This setting may be configured in your .vimrc file (if it is sourced).
2. I can't reproduce the issue on my Windows machine. Does the `dd` command always override your system clipboard, or are there specific steps to reproduce this behavior?

**Reply by widavies** [reply]
Thanks, all it took was `set clipboard=ideaput` without the '+'.

Out of curiosity, the ideal behavior I'm going for is to link my system keyboard to VIM with `unnamed`, so it would work like this:
- Copy some text from my web browser
- Use `dd` to delete some text
- Unnamed register now has `dd`'d text, but ideally, `"0p` or `"*p` would still contain the web browser text
- If I didn't do a `dd`, then pressing just `p` (no leading keys) would also still contain the system clipboard text

Typically what happens is that `dd` will delete the text and overwrite the unnamed register with what it just deleted.

I know this is how VIM works, and that's fine. However, in this scenario, the last yank register `"0` does not get affected by copying text from the web browser. So after doing `dd`, the text from the web browser is gone (when using `unnamed`). Is there a way to make system clipboard text also go into the `"0` register or somewhere where it wouldn't disappear?
